### ISSUE TITLE
SSI DIM Wallet Stub can't create Industry Core Policy

### ISSUE THREAD

**Issue by evegufy** [eclipse-tractusx/tractus-x-umbrella#174]
### Current Behavior

@matbmoser @ds-jhartmann Could you please provide more information?

**Comment by evegufy** [comment]
@matbmoser @ds-jhartmann Once the description is ready. Please add Nitin, maybe also discuss in the Umbrella Matrix Chat.

Should not be a blocker for the community. Just use empty policies for now.

**Comment by matbmoser** [comment]
This is related with the policy, because the access policies will be checked against the wallet, and we usually configure the same policy for access and usage.

Why is the stub not accepting things by default???

**Comment by ds-jhartmann** [comment]
@evegufy I added more details regarding policies. 
@matbmoser please add additional information in case I forgot something or got something wrong

**Comment by evegufy** [comment]
@ds-jhartmann the following error you sent via mail:

`
SEVERE 2024-11-25T09:42:01.416944841 JerseyExtension: Unexpected exception caught
 java.lang.NullPointerException: Cannot invoke "Object.toString()" because "v" is null
 at org.eclipse.edc.iam.identitytrust.service.IdentityAndTrustService.lambda$obtainClientCredentials$0(IdentityAndTrustService.java:114)
`

Is it related to this issue?

**Comment by ds-jhartmann** [comment]
@evegufy no, this is not related to the policies

**Comment by evegufy** [comment]
@nitin-vavdiya could you please give your option on this topic?

**Comment by nitin-vavdiya** [comment]
Do we have an error log from the SSI stub application? 

Do we have what request body is being sent to stub application in the query presentation API call? We are printing the request body in the logs in SSI stub application.

At the current stage, SSI stub application supports only three types of credentials:

1. MembershipCredential
2. DataExchangeGovernanceCredential
3. BpnCredential

If the requested type does not exist in this list, then it will give an error.

For ref:

![image](https://github.com/user-attachments/assets/7e7405a1-2a65-4fd3-b73d-1cbaa4d261cb)

**Comment by evegufy** [comment]
@nitin-vavdiya thanks for the quick response!
@matbmoser @ds-jhartmann could you please answer Nitin's questions?

**Comment by matbmoser** [comment]
@nitin-vavdiya why the usage purpose are not allowed?

**Comment by nitin-vavdiya** [comment]
Thanks @matbmoser 

Can you please provide a schema/example for these credentials? We need this to prepare credential subject.

**Comment by matbmoser** [comment]
@nitin-vavdiya please enable this two constraints in the policy definition, so that the wallet does not gives an error for them.
Thanks, if you need help tell me ;)

**Comment by nitin-vavdiya** [comment]
@matbmoser  

Thanks for replay.
Wallet stub application neither creating any kind of policy nor doing policy validation in during contract negotiations. This done in EDC control plane. There is no role of wallet stub application.

**Comment by lgblaumeiser** [comment]
Is there any more log information on what is happening. Actually, @nitin-vavdiya is right, if it fails due to the UsagePurpose, the wallet is not involved. What happens, if the policy only contains one of the two elements, is it accepted?

**Comment by nitin-vavdiya** [comment]
@evegufy @matbmoser @ds-jhartmann 
We need to assign this ticket to the right people for better and quick resolution.

**Comment by matbmoser** [comment]
> Is there any more log information on what is happening. Actually, [@nitin-vavdiya](https://github.com/nitin-vavdiya) is right, if it fails due to the UsagePurpose, the wallet is not involved. What happens, if the policy only contains one of the two elements, is it accepted?

this is not true, because the wallet stub, should accept any usage purpose. -> I am not sure if the stub does that.

if it was failing because of the traceability framework agreement, thats fine, since it is deprecated.
